### DISCUSSION TITLE
"basic" addons ?

### DISCUSSION THREAD

**Discussion by icf20** [OCA/odoo-community.org#174] [Q&A]
is there any curated list with "basic" addons ? oca having 200 repos kinda make searching hard

**Comment by antoniospneto** [comment]
You can consult in a unified way here: https://odoo-community.org/shop

**Reply by icf20** [reply]
yes and no for example if i want to activate l10n_es_vat_book i got 4 levels deep dependencies, so seeing how much stuff depends on report_xlsx would kinda make report_xlsx basic. Sorting the shop by maturity and downloads kinda do something but the downloads numbers are unreliable

**Reply by antoniospneto** [reply]
@icf20 I did not understand your question, sorry if I can't help.

**Reply by antoniospneto** [reply]
I have seen tools in some places that create a dependency graph, but I don't remember the links. Maybe someone else in the community can help with that.

**Comment by vdewulf** [comment]
Hello,
With the consultants OCA workgroup, we have a list of "must have" modules but it's based on the experiences from consultants over the past years: several OCA modules are always installed. This list is under creation on our website (not published yet).

If you're question is about basic modules dependency, I don't know if it's really necessary to have this list because the modules will get installed automatically upon installation of the module you need, provided that you have the corresponding code on your server.

Is it the issue you're encountering, loosing time to install modules depending on modules in other repos?

**Reply by icf20** [reply]
> loosing time

usually i want to test a module so I have to do

1 get the code from github
2 upload on server 
3 restart odoo
4 go to app list
5 check what is missing to install
do 1,2,3,4,5 steps until can finally install

**Reply by vdewulf** [reply]
Hello,

Thanks for clarifying your context.

What I usually do to test an OCA module is:
1. Test it on the runboat directly (not on my server), so that I don't need to install anything, just clic and test. 
Ref blog post about this tool: 
https://odoo-community.org/blog/news-updates-1/goodbye-runbot-welcome-runboat-116

How to use the runboat explained here (step 4 at the bottom): 
https://odoo-community.org/resources/review 

2. Once validated on the runboat, I want to test it with my customers database on my test server. I'm not a dev profile so I asked my colleagues to do it, but I think they checked on the Readme file, the section "depends", to help find the prerequisites, like for the module you mentionned here:
https://github.com/OCA/l10n-spain/blob/924ea91e16eb5921d2b54d1ee0bd1821384e2257/l10n_es_vat_book/__manifest__.py#L18

Some thouthful contributors even mention the corresponding repos where the dependencies can be find, like this example on MIS Builder: 
https://github.com/OCA/mis-builder/blob/3aa42b4b03a2267147efb45125325e194f8a8aaf/mis_builder/__manifest__.py#L13

Some repos also have a REQUIREMENT.txt doc explaining about python libraries dependencies for the whole repo. Example here on the server-auth repo:
https://github.com/OCA/server-auth/blob/16.0/requirements.txt

This is in fact all explained in the guidelines about contribution here:
https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#external-dependencies

In conclusion, there should be info to help you not loose too much time, if the contribution guidelines are well followed!

Hope you'll find your way in the OCA, don't hesitate to share this type of question on the Discord channel as well (invitation link available on this page: https://www.odoo-community.org/qrlinks). Others might have other ideas to help you on this specific question.

Have a nice day and a simple "thank you" from you is always nice to receive when we take time to answer your questions!
